kubernetes/kubernetes_test.go

```diff
@@ -155,6 +156,7 @@ func TestCreateAgentStatusMetric(t *testing.T) {
 	})
 }
 
+//nolint gocyclo
```

Comment:
Any way we could simplify this function a bit by breaking it out into some other functions? I think that would be preferable to adding these comments